go/libraries/doltcore/env/remotes.go

```diff
@@ -372,6 +377,19 @@ func NewPullSpec(ctx context.Context, rsr RepoStateReader, remoteName string, sq
 		return nil, ErrNoRefSpecForRemote
 	}
 
+	trackedBranches, err := rsr.GetBranches()
+	if err != nil {
+		return nil, err
+	}
+
+	if _, hasUpstream := trackedBranches[branch.GetPath()]; !hasUpstream {
+		if remoteOnly {
+			return nil, fmt.Errorf("You asked to pull from the remote '%s', but did not specify a branch. Because this is not the default configured remote for your current branch, you must specify a branch on the command line.", remoteName)
```

Comment:
(minor) Would be nice to throw this into a named Err var, too, just like the Err var below. Maybe called `ErrPullWithRemoteNoUpstream`?